File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) == 0 : return 0
        nums[:] = sorted(nums)
        prev_num , max_len , count = float('-inf') , 1 , 0

        
        for num in nums :
            if num - 1 == prev_num :
                count += 1
                prev_num = num
                
            elif num != prev_num :
                count = 1
                prev_num = num
            
            max_len = max(max_len , count)
            
        return max_len

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        
        if(nums.length == 1) return 1;
        if(nums.length == 0) return 0;
        Arrays.sort(nums);
        int count = 1;
        int max_len = count;
        

        for(int i=1;i<nums.length;i++){
            while(i < nums.length && nums[i] == nums[i-1]) i++;
            if(i < nums.length && nums[i] == nums[i-1] + 1){
                count++;
                max_len = Math.max(max_len , count);
            }
            else{
                count = 1;
            }
        }
        
        return max_len;
    }
}